### PR TITLE
Added support for custom exception handler for VertxPrometheusOptions…

### PR DESCRIPTION
**Pull request by swamymavuri**
… (#213)

Motivation:

Explain here the context, and why you're making that change, what is the problem you're trying to solve.

Conformance:

Your commits should be signed and you should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

**Review by tsegismont — COMMENTED** [review]
Thanks for your proposal. Instead of adding an exception handler to options, it would be better to create a class that provides a request handler (similar to the scraping handler for vert.x web).
This way, the user can create the HTTP server without implementing the request handler and having full control on the server behavior.

**Comment by swamymavuri** [comment]
Hi @tsegismont ,
Could you please review the following code changes as per your suggestions

Created a PrometheusRequestHandler class to manage Prometheus requests.
By default, PrometheusRequestHandler  is used while creating HttpServer if embeddedServer is enabled.

and User can get the flexibility to create their own httpServer without implementing the request handler

Thanks

**Comment by swamymavuri** [comment]
> Can you please sign the Eclipse Contributor Agreement and then, using the same Git author/user email, sign-off the commit?

Thanks @tsegismont , revalidated the commit by signing off the agreement.

File: src/main/java/io/vertx/micrometer/PrometheusRequestHandler.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * A request handler for serving Prometheus metrics.
+ *
+ * @author Swamy Mavuri
+ */
+public class PrometheusRequestHandler implements Handler<HttpServerRequest> {
+
+  private final PrometheusMeterRegistry registry;
+  private final VertxPrometheusOptions options;
```
Comment (tsegismont) [inline src/main/java/io/vertx/micrometer/PrometheusRequestHandler.java:17]: Can you change this to a `String` field that stores the embedded server endpoint? We don't need all of the options.

File: src/main/java/io/vertx/micrometer/PrometheusRequestHandler.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+   */
+  public PrometheusRequestHandler(PrometheusMeterRegistry registry, VertxPrometheusOptions options) {
+    this.registry = registry;
+    this.options = options;
+  }
+
+
+  @Override
+  public void handle(HttpServerRequest request) {
+    if (options.getEmbeddedServerEndpoint().equals(request.path())) {
```
Comment (tsegismont) [inline src/main/java/io/vertx/micrometer/PrometheusRequestHandler.java:33]: Here you can simplify the condition to use the string field

File: src/main/java/io/vertx/micrometer/PrometheusRequestHandler.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerRequest;
+
+/**
+ * A request handler for serving Prometheus metrics.
+ *
+ * @author Swamy Mavuri
+ */
+public class PrometheusRequestHandler implements Handler<HttpServerRequest> {
```
Comment (tsegismont) [inline src/main/java/io/vertx/micrometer/PrometheusRequestHandler.java:14]: Can you please create an interface annotated with `@VertxGen` like the `PrometheusScrapingHandler`

```
@VertxGen
public interface PrometheusScrapingHandler {

  /**
   * Creates a Vert.x Web {@link io.vertx.ext.web.Route} handler for Prometheus metrics scraping.
   * The default backend registry is used.
   *
   * @return a {@link io.vertx.ext.web.Route} handler for the default backend registry
   * @see BackendRegistries#getDefaultNow()
   */
  static Handler<RoutingContext> create() {
    return new PrometheusScrapingHandlerImpl();
  }
```

If you do this, we will automatically have generated files for RxJava, Mutiny, ...etc

**Review by tsegismont — COMMENTED** [review]
Thanks for the updates @swamymavuri

**Comment by tsegismont** [comment]
Can you please also change the PR and issue titles and descriptions to reflect the new direction?

**Pull request closed without merging**